File: src/led_flashing.cpp
#include "ros/ros.h"
#include "std_msgs/Bool.h"

int main(int argc, char **argv){
  ros::init(argc, argv, "led_flashing");
  ros::NodeHandle nh;
  ros::Publisher publisher = nh.advertise<std_msgs::Bool>("/publisher_tutorial/led_state", 1);
  ros::Rate sleep_time(1);

  bool state;
  std_msgs::Bool msg;

  while (ros::ok()){
    ROS_INFO("ON");
    state = true;
    msg.data = state;
    publisher.publish(msg);
    sleep_time.sleep();

    ROS_INFO("OFF");
    state = false;
    msg.data = state;
    publisher.publish(msg);
    sleep_time.sleep();
  }

  return 0;

}

File: src/led_flashing.py
#! /usr/bin/env python
# coding:utf-8

import rospy
import time
from std_msgs.msg import Bool

if __name__ == "__main__":
    rospy.init_node("led_flashing", anonymous=True)
    publisher = rospy.Publisher("/publisher_tutorial/led_on", Bool, queue_size=1)

    while not rospy.is_shutdown():
        try:
            publisher.publish(True)
            time.sleep(1)

            publisher.publish(False)
            time.sleep(1)

        except KeyboardInterrupt: #Ctrl+cが入力された
            print "Ctrl+c"
            break
